### DISCUSSION TITLE
See logging of successful transmission

### DISCUSSION THREAD

**Discussion by maxbobse** [PHPMailer/PHPMailer#2740] [Q&A] [answered]
Is there a way to enable the debug output or logging for successful transmissions?
Problem is that I am sending mails to sendinblue and PHPMailer shows the transmission as success but nothing shows up on their end. So I would like to provide them with some more detailed logging information to prove that it seems to work from my side.

**Comment by Synchro** [comment]
Set `$mail->SMTPDebug = 2` and it will show the whole SMTP conversation, hopefully including a `250 OK` response to your `DATA` command; that's what you need to show them to prove that it's accepting your messages. Onward delivery from there is not up to PHPMailer.

**Comment by maxbobse** [comment]
I already have that but still just get the 'Message has been sent' in the error log. Code below:

```
$this->mailer = new PHPMailer(true);

        try {
            $this->mailer->SMTPDebug = 2;

            $mail->DebugOutput = function ($str, $level) {
                error_log($str);
            };

            $this->mailer->isSMTP();                                            //Send using SMTP
            $this->mailer->Host = 'smtp-relay.sendinblue.com';                     //Set the SMTP server to send through
            $this->mailer->SMTPAuth = true;                                   //Enable SMTP authentication
            $this->mailer->Username = 'xxx@yyy.com';                     //SMTP username
            $this->mailer->Password = 'myLittlePassword';                               //SMTP password
            $this->mailer->SMTPSecure = PHPMailer::ENCRYPTION_STARTTLS;            //Enable implicit TLS encryption
            $this->mailer->Port = 587;                                    //TCP port to connect to; use 587 if you have set `SMTPSecure = PHPMailer::ENCRYPTION_STARTTLS`
            $this->mailer->SMTPDebug = 2;

            //Recipients
            $this->mailer->setFrom($mail->sender);

            foreach ($mail->receiversTO as $key => $value) {
                $this->mailer->addAddress($value, '');     //Add a recipient    
            }

            //Content
            $this->mailer->isHTML($mail->isHTML);                                  //Set email format to HTML
            $this->mailer->Subject = $mail->subject;
            $this->mailer->Body = $mail->body;

            $this->mailer->send();
            error_log('Message has been sent');
        } catch (Exception $e) {
            error_log("Message could not be sent. Mailer Error: " . $this->mailer->ErrorInfo);
        }
```

Am I missing anything else here?

**Reply by Synchro** [reply]
Simple mistake:

```php
            $mail->DebugOutput = function ($str, $level) {
                error_log($str);
            };
```

should be:

```php
            $this->mailer->DebugOutput = function ($str, $level) {
                error_log($str);
            };
```

**Reply by maxbobse** [reply]
OMG ... thanks for spotting that! Made the change but I am still not seeing anything new in the error log file. :/

**Reply by Synchro** [reply]
I'd just skip setting `DebugOutput` and copy and paste the default output. You only need this to work once, so no point in going to any effort. FWIW, `DebugOutput` can send output to the error log anyway if you set it to `error_log`:

    $this->mailer->DebugOutput = 'error_log';

**Reply by maxbobse** [reply]
Ok, getting closer here ... just realised that the logging was always echoed back so it got passed through in my PHP script to the initial request, at least now I can see the logging :)

Only question remains: why doesn't it write to the error log? I have added

```
$this->mailer->DebugOutput = 'error_log';
```
and removed the DebugOutput function piece but still no luck.

**Reply by maxbobse** [reply]
Found the issue ... variable in class PHPMailer is called Debugoutput and not DebugOutput. Now its working fine :)

**Comment by maxbobse** [comment, marked as answer]
Found the issue ... variable in class PHPMailer is called Debugoutput and not DebugOutput

**Reply by Synchro** [reply]
Doh!
